### ISSUE TITLE
G John server error

### ISSUE THREAD

**Issue by ctschroeder** [CopticScriptorium/cts#136]
Hi @amir-zeldes. I got the following email this week at the public info email address for the project.
> Dears,
> 
> The following link is not working and I think that all the gospel of John is not working.
> http://data.copticscriptorium.org/texts/new-testament/43_john_1/sahidica
> 
> --
> Regards,
> Fady

It indeed appears there is a "server error" for most (all?) G John documents.  I'm not sure what's going on there.  Can someone look into it?  Thanks so very much!

**Comment by lgessler** [comment]
Took a glance, looks like the issue's that this document doesn't have a `document_cts_urn` metadatum. From the corresponding .tt file:

```
<meta corpus="sahidica.nt" license="&lt;a href='http://www.copticscriptorium.org/download/corpora/Mark/coptic_nt_sahidic.html'&gt;(c)2000-2006 by J Warren Wells, for academic use only&lt;/a&gt;." source="Sahidica - A New Edition of the New Testament in Sahidic Coptic" annotation="Caroline T. Schroeder, Amir Zeldes" version_date="2018-04-01" version_n="2.5.0" translation="World English Bible (WEB)" languages="Sahidic Coptic" project="Coptic SCRIPTORIUM"  chapter="1" title="43_John_1">
```

Should every document have a `document_cts_urn`, or does the application need to accommodate documents without a `document_cts_urn`?

**Comment by ctschroeder** [comment]
Yes. Since this is an automated corpus, all of the document urns were are generated at the same time for the whole Sahidic NT. I can check Gitdox

**Comment by amir-zeldes** [comment]
I just went through the NT scripts, looks like the identifier for John was broken, prob. some sort of copy paste error due to 1 John. So these should be just .john and 1 John etc. should stay `cts:copticLit:nt.1john` right?

**Comment by lgessler** [comment]
Revelation's also missing them, but that appears to be all for the NT and OT:

```
>>> for fn in sorted(fnames):
...     s = open(fn, 'r').read()
...     if 'document_cts_urn' not in s:
...             print(fn)
... 
43_John_01.tt
43_John_02.tt
43_John_03.tt
43_John_04.tt
43_John_05.tt
43_John_06.tt
43_John_07.tt
43_John_09.tt
43_John_10.tt
43_John_11.tt
43_John_12.tt
43_John_13.tt
43_John_14.tt
43_John_15.tt
43_John_16.tt
43_John_17.tt
43_John_18.tt
43_John_19.tt
43_John_20.tt
43_John_21.tt
66_Revelation_01.tt
66_Revelation_02.tt
66_Revelation_03.tt
66_Revelation_04.tt
66_Revelation_05.tt
66_Revelation_06.tt
66_Revelation_07.tt
66_Revelation_08.tt
66_Revelation_09.tt
66_Revelation_10.tt
66_Revelation_11.tt
66_Revelation_12.tt
66_Revelation_13.tt
66_Revelation_14.tt
66_Revelation_15.tt
66_Revelation_16.tt
66_Revelation_17.tt
66_Revelation_18.tt
66_Revelation_19.tt
66_Revelation_20.tt
66_Revelation_21.tt
66_Revelation_22.tt
```

**Comment by amir-zeldes** [comment]
Thanks to both of you! I just hotfixed this in GitHub and re-ingested. It still needs to be propagated to all format versions of the NT, but I've laid the groundwork for that to happen on the next release, which I'm planning to do with added auto-NER annotations. The CTS repo should be working correctly now.

**Comment by ctschroeder** [comment]
Yes, John is gospel of John. 1John, 2John, 3John would be letters of John.
Thanks for taking care of this. I realized Saturday this was not something I could fix in GitDox and was planning to check in with you both today.
Do we have a csv file with these and the OT info for future reference?  I feel like this has happened before.

**Comment by amir-zeldes** [comment]
Yes, it's definitely something I needed to fix in the NT pipeline. There's no CSV file, it's part of a python script that isn't really committed anywhere... I could try to add it somewhere, but it was honestly never meant to be invoked again (just the first time when we did the split into individual documents, and auto generated the metadata).

The script has been fixed, as has the output data, which is also not something we've been committing. That data gets auto-processed using the NLP tools for each new release, and the output of that is what ends up on GH. I could just add this to bible-dev I guess? But it probably won't run out of the box without setting up the environment, so mainly just for archival purposes.
